library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+- (instancetype)initWithConfig:(NSString *)config {
+  self = [super init];
+  if (self) {
+    self.runner = [[NSThread alloc] initWithTarget:self selector:@selector(run:) object:config];
+    [self.runner start];
+  }
+  return self;
+}
+
+- (BOOL)isRunning {
+  return self.runner.isExecuting;
+}
+
+- (BOOL)isTerminated {
+  return self.runner.isFinished;
```

Comment:
This one we might want `self.runner == nil || self.runner.isFinished` I think?

Reply:
yep, good catch

Reply:
Oh actually, I think the current logic is fine - Terminated is only true if it's gone through all other lifecycle phases and then stopped. So `runner` being `nil` means it hasn't started yet, in other words, it's not terminated. (I think eventually we can formalize this all better.)

Reply:
i.e. the current logic is the same as what was originally there; it just takes advantage of objective-c's nil-dispatch behavior.